### PR TITLE
Makefile.am: Fix underlinking

### PR DESCRIPTION
**Pull request by sardemff7**

### CODE REVIEW

File: tests/gegl/Makefile.am [hunk 1]
```diff
@@ -29,8 +29,8 @@ endif
 LDADD = \
 	$(DEPS)								\
 	$(GEGL_LIBS)						\
+	$(top_builddir)/tests/libmypaint-tests.a \
 	$(top_builddir)/libmypaint.la		\
-	$(top_builddir)/gegl/libmypaint-gegl.la	\
-	$(top_builddir)/tests/libmypaint-tests.a
+	$(top_builddir)/gegl/libmypaint-gegl.la
```
Comment (achadwick) [inline tests/gegl/Makefile.am]: Does ordering matter here?

Comment (sardemff7) [inline tests/gegl/Makefile.am]: Yes. Try adding `-Wl,--no-undefined` to your `LDFLAGS’ and test with and without this patch. :-)

**Comment by achadwick** [comment]
Hi there - what underlinking problem does this solve? We already use `LIBS = @LIBS` since 78ac56200d37b8a0bf022bf6722e9ac67a742df5 and `-no-undefined` since bf3cd0c450eb245e5b7988de57a5987c244d0496.

    $ ldd -r .libs/libmypaint.so >/dev/null
    [no output]

But that's only the "direct" case, of course.

**Comment by sardemff7** [comment]
Here are the logs:
[without the patch](https://github.com/mypaint/libmypaint/files/306664/log1.txt)
[with the first part (without the reordering)](https://github.com/mypaint/libmypaint/files/306665/log2.txt)
I use `-Wl,--as-needed` too, and note that `-no-undefined` is not very strong, libtool often ignore it.

You’re right about `$(LIBS)` but it is good practice to avoid putting everything in that an relying on the default values. `*_LIBADD` (and not `*_LIBS`, nor `*_LDADD` since you are building  libs here) is cleaner.

Also, `LIBS` contains `-lm` as you use `AC_SEARCH_LIBS`.

I started an Autotools usage cleanup, but you do use a lot of shortcuts, so a simple fix will do for now. :-)

File: tests/gegl/Makefile.am (hunk 1 above)
Comment (QuLogic) [inline tests/gegl/Makefile.am]: Strictly speaking, shouldn't `$(GEGL_LIBS)` be last?

Comment (sardemff7) [inline tests/gegl/Makefile.am]: Oh, strictly speaking, yes.

**Pull request merged**